src/wp-admin/authorize-application.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Authorize Application Screen
+ *
+ * @package WordPress
+ * @subpackage Administration
+ */
+
+/** WordPress Administration Bootstrap */
+require_once __DIR__ . '/admin.php';
+
+if ( isset( $_POST['action'] ) && 'authorize_application_password' === $_POST['action'] ) {
+	check_admin_referer( 'authorize_application_password' );
+
+	$success_url = $_POST['success_url'];
+	$reject_url  = $_POST['reject_url'];
+	$app_name    = $_POST['app_name'];
+	$redirect    = admin_url();
+
+	if ( isset( $_POST['reject'] ) ) {
+		if ( $reject_url ) {
+			// Explicitly not using wp_safe_redirect b/c sends to arbitrary domain.
+			$redirect = esc_url_raw( add_query_arg( 'success', 'false', $reject_url ) );
+		}
+	} elseif ( isset( $_POST['approve'] ) ) {
+		list( $new_password, $new_item ) = WP_Application_Passwords::create_new_application_password( get_current_user_id(), $app_name );
+		if ( empty( $success_url ) ) {
+			wp_die( '<h1>' . __( 'Your New Application Password:' ) . '</h1><h3><kbd>' . esc_html( self::chunk_password( $new_password ) ) . '</kbd></h3>' );
```

Comment:
Could we make the no-js fallback re-render the page instead of using `wp_die` to generate output?

Reply:
Made this change.